apps/backend/src/libs/modules/http/base-http.module.ts

```diff
@@ -0,0 +1,55 @@
+import { type ValueOf } from "~/libs/types/types.js";
+
+import { HTTPCode, HTTPError, type HTTPOptions } from "./libs/enums/enums.js";
+import { HTTP } from "./libs/types/types.js";
+
+class BaseHTTP implements HTTP {
+	private checkStatus = async (response: Response): Promise<Response> => {
+		if (!response.ok) {
+			const parsedException = (await response.json().catch(() => ({
+				message: response.statusText,
+			}))) as Record<"message", string>;
+
+			throw new HTTPError({
+				message: parsedException.message,
+				status: response.status as ValueOf<typeof HTTPCode>,
+			});
+		}
+
+		return response;
+	};
+
+	private getUrl = <T extends Record<string, unknown>>(
+		url: string,
+		query: T | undefined,
+	): string => {
+		if (query) {
+			const newUrl = new URL(url);
+			for (const [key, value] of Object.entries(query)) {
+				newUrl.searchParams.set(key, value as string);
+			}
```

Comment:
```suggestion
			const queryParameters = new URLSearchParams(query);
```
Then use `queryParameters.toString()`